authority/voting/server/server.go

```diff
@@ -200,24 +204,69 @@ func New(cfg *config.Config) (*Server, error) {
 	}
 
 	// Initialize the authority identity key.
-	var err error
 	identityPrivateKeyFile := filepath.Join(s.cfg.Authority.DataDir, "identity.private.pem")
 	identityPublicKeyFile := filepath.Join(s.cfg.Authority.DataDir, "identity.public.pem")
-	if s.identityKey, err = eddsa.Load(identityPrivateKeyFile, identityPublicKeyFile, rand.Reader); err != nil {
-		s.log.Errorf("Failed to initialize identity key: %v", err)
-		return nil, err
+
+	s.identityPrivateKey, s.identityPublicKey = cert.Scheme.NewKeypair()
+	var err error
+	if pem.BothExists(identityPrivateKeyFile, identityPublicKeyFile) {
+		err = pem.FromFile(identityPrivateKeyFile, s.identityPrivateKey)
+		if err != nil {
+			return nil, err
+		}
+		err = pem.FromFile(identityPublicKeyFile, s.identityPublicKey)
+		if err != nil {
+			return nil, err
+		}
+	} else if pem.BothNotExists(identityPrivateKeyFile, identityPublicKeyFile) {
+		err = pem.ToFile(identityPrivateKeyFile, s.identityPrivateKey)
+		if err != nil {
+			return nil, err
+		}
+		err = pem.ToFile(identityPublicKeyFile, s.identityPublicKey)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("%s and %s must either both exist or not exist", identityPrivateKeyFile, identityPublicKeyFile)
 	}
 
-	scheme := wire.NewScheme()
+	scheme := wire.DefaultScheme
 	linkPrivateKeyFile := filepath.Join(s.cfg.Authority.DataDir, "link.private.pem")
 	linkPublicKeyFile := filepath.Join(s.cfg.Authority.DataDir, "link.public.pem")
-	if s.linkKey, err = scheme.Load(linkPrivateKeyFile, linkPublicKeyFile, rand.Reader); err != nil {
-		s.log.Errorf("Failed to initialize link key: %v", err)
-		return nil, err
+
+	var linkPrivateKey wire.PrivateKey = nil
+	var linkPublicKey wire.PublicKey = nil
+
+	if pem.BothExists(linkPrivateKeyFile, linkPublicKeyFile) {
```

Comment:
I prefer to not have keyfiles external to the configuration file, and do not feel that the serialized publickeys are large enough to warrant this change. Can we just use the Marshaller interface to serialize PEM encoded keys from the Config fields?